cloud-aws-native/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/resource/instance/AwsNativeEIPResourceBuilder.java

```diff
@@ -109,25 +110,28 @@
             Optional<CloudResource> instanceResourceOpt = persistenceRetriever.notifyRetrieve(ac.getCloudContext().getId(), String.valueOf(privateId),
                     CommonStatus.CREATED, ResourceType.AWS_INSTANCE);
             CloudResource instanceResource = instanceResourceOpt.orElseThrow();
-            List<String> eips = List.of(allocateAddressResult.getAllocationId());
+            String allocationId = allocateAddressResult.getAllocationId();
+            List<String> eips = List.of(allocationId);
+            String instanceId = instanceResource.getInstanceId();
             List<AssociateAddressResult> associateAddressResults = awsElasticIpService.associateElasticIpsToInstances(
                     amazonEC2Client,
                     eips,
-                    List.of(instanceResource.getInstanceId()));
+                    List.of(instanceId));
 
+            String associationId = associateAddressResults.get(0).getAssociationId();
             CloudResource cloudResource = CloudResource.builder()
                     .cloudResource(buildableResource.get(0))
-                    .persistent(true)
-                    .instanceId(instanceResource.getInstanceId())
-                    .reference(allocateAddressResult.getAllocationId())
+                    .instanceId(instanceId)
+                    .status(CommonStatus.CREATED)
+                    .reference(allocationId)
                     .params(Map.of(CloudResource.ATTRIBUTES, EIpAttributes.EIpAttributesBuilder.builder()
-                            .withAllocateId(allocateAddressResult.getAllocationId())
-                            .withAssociationId(associateAddressResults.get(0).getAssociationId())
+                            .withAllocateId(allocationId)
+                            .withAssociationId(associationId)
                             .build()))
                     .build();
             ret.add(cloudResource);
-            LOGGER.info("EIp created for {}: association id: {}, allocationId: {}",
-                    cloudInstance.getInstanceId(), associateAddressResults.get(0).getAssociationId(), allocateAddressResult.getAllocationId());
+            LOGGER.info("EIp created for {}: association id: {}, allocationId: {}", instanceId, associationId,
+                    allocationId);
```

Comment:
I wonder if you also meant to insert `instance with privateId:`  before the first `{}` and replace `instanceId` with `privateId`, similarly to line 97.